src/readinglistbar.cpp

```diff
@@ -31,12 +31,15 @@ void ReadingListBar::setupList()
     auto listWidget = ui->listWidget;
     listWidget->clear();
     for(auto& bookmark:bookmarks) {
-        auto reader = library->getReader(QString::fromStdString(bookmark.getBookId()));
-        if (reader == nullptr)
+        std::shared_ptr<zim::Archive> archive;
+        try {
+            archive = library->getArchive(QString::fromStdString(bookmark.getBookId()));
+        } catch (std::out_of_range& e) {
             continue;
-        std::string content;
-        std::string mimeType;
-        reader->getFavicon(content, mimeType);
+        }
+        auto illustration = archive->getIllustrationItem(48);
```

Comment:
How do you handle it?
Currently, I have just left the catch block empty.

Reply:
It depends of the use case.
We probably want to display nothing (in the library or in the tab icon) if the zim has no illustration. It seems that lefting the catch block empty is enough.
(In the library we display a "broken" image. It would be better to not display an image at all. But it is not the purpose of this PR)